### PR TITLE
Using the services methods from fip-commons

### PR DESCRIPTION
**Pull request by nandajavarma**

### CODE REVIEW

File: examples/redis-deployment/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2021 SIGHUP s.r.l All rights reserved.
```
Comment (angelbarrera92) [inline examples/redis-deployment/namespace.yaml:1]: Is this intended?

File: internal/config/config.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	LogLevel    string
+	MinEp       int
+	Namespace   string
+	ServiceName string
+	KubeConfig  string
+	CfgFile     string
+}
+
+// NewSecConfig creates an empty configuration struct.
+func NewSecConfig() *SecConfig {
```
Comment (angelbarrera92) [inline internal/config/config.go:21]: If you are exposing a constructor... Is it required to expose the `SecConfig` type?
In other words...
what happens if you rename the `SecConfig` type to `secConfig`?

File: internal/services/main_test.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) 2021 SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package services
```
Comment (angelbarrera92) [inline internal/services/main_test.go:5]: If the file has no content, feel free to remove it

File: cmd/service-endpoints-check/main.go [hunk 4]
```diff
@@ -5,58 +5,60 @@
...
-		log.Infof("Service %v of namespace %v has requested number of endpoints(%v)",
-			serviceName, namespace, minEp)
-		os.Exit(0)
+var rootCmd = &cobra.Command{ //nolint:gochecknoglobals
+	PersistentPreRunE: cmdConfig,
+	Use:               "service-endpoints-check",
+	Short:             "Health check to monitor the endpoints of a service",
+	SilenceUsage:      true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := context.Background()
```
Comment (angelbarrera92) [inline cmd/service-endpoints-check/main.go:36]: Instead of using a new context (background), try to use the cobra context. Something like `cmd.Context()`

**Review by angelbarrera92 — CHANGES_REQUESTED** [review]
Overall looks good, just take a look to the comments i left in the PR

File: examples/redis-deployment/namespace.yaml (hunk 1 above)
Comment (nandajavarma) [reply examples/redis-deployment/namespace.yaml:1]: nope. fixed. 🤦

File: internal/config/config.go (hunk 2 above)
Comment (nandajavarma) [reply internal/config/config.go:21]: SecConfig is imported in other files. To define type. like `var cfg *config.SecConfig`

File: cmd/service-endpoints-check/main.go (hunk 4 above)
Comment (nandajavarma) [reply cmd/service-endpoints-check/main.go:36]: oooh great point! still getting my head around contexts! will do

File: internal/services/main_test.go (hunk 3 above)
Comment (nandajavarma) [reply internal/services/main_test.go:5]: done!

File: internal/config/config.go (hunk 2 above)
Comment (angelbarrera92) [reply internal/config/config.go:21]: What about:

`cfg := NewSecConfig()`

Comment (nandajavarma) [reply internal/config/config.go:21]: we have to still use SecConfig  outside as a type. Marking it as resolved.

**Pull request merged**